### DISCUSSION TITLE
The JDK Flight Recorder File Format - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#292] [Announcements]
# The JDK Flight Recorder File Format - Gunnar Morling

The JDK Flight Recorder (JFR) is one of Java’s secret weapons;
deeply integrated into the Hotspot VM, it’s a high-performance event collection framework,
which lets you collect metrics on runtime aspects like object allocation and garbage collection,
class loading, file and network I/O, and lock contention, do method profiling, and much more.

JFR data is persisted in recording files
(since Java 14, also "realtime" event streaming is supported),
which can be loaded for analysis into tools like JDK Mission Control (JMC),
or the jfr utility coming with OpenJDK itself.

https://www.morling.dev/blog/jdk-flight-recorder-file-format/

**Comment by thegreystone** [comment]
Hi Gunnar!

The "formal" name for the events with the constant pool data is check point events (jdk.CheckPoint) - just mentioning since people might be using the jfr command line tool and wonder what those events are.

Kind regards,
Marcus

**Reply by gunnarmorling** [reply]
Hey Marcus, thanks for reading and commenting!

>  people might be using the jfr command line tool and wonder what those events are.

That's very interesting; how do you see these events using the _jfr_ command? I only see the actual data events, but neither the metadata event nor those checkpoint events? Is there some flag for getting these too?

**Reply by thegreystone** [reply]
Try jfr summary.

**Reply by thegreystone** [reply]
Also, you can look at the constant pools in JMC 8.2+.

**Reply by gunnarmorling** [reply]
Thanks, I've updated the image, mentioning that `jfr summary` will display the number of checkpoint events.
